src/test/java/org/cy3sbml/miriam/RegistryUtilTest.java

```diff
@@ -23,8 +93,72 @@ public void updateMiriamXML() throws Exception {
 
 
     @Test
-    public void loadRegistry() throws Exception {
+    public void loadRegistry() {
         assertNotNull(RegistryUtil.getMiriamContent());
     }
 
-}
\ No newline at end of file
+    @ParameterizedTest
+    @MethodSource("resourceFilesProvider")
+    public void testResourceUriProcessing(String resourceURI) throws IOException {
+
+
+            String[] tokens = resourceURI.split("/");
+            //created a checkednamespaces list so the code skips the namespaces it already checked before (instead of checking for thousands of identifiers links)
+            String checkedNamespace;
+            if (tokens[3].contains(":")){
+                checkedNamespace = tokens[3].split(":")[0];
+            } else {
+                checkedNamespace = tokens[3];
+            }
+            // added brackets around the names in case some namespaces contain others e.g. "go" and may be there is a namespace such as "goxyz"
+            if (!checkedNamespaces.contains("[" + checkedNamespace + "]")) {
+                checkedNamespaces.add("[" + checkedNamespace + "]");
+
+                //need to eliminate the invalid identifiers links
+                URL url = new URL(resourceURI);
+                HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+
+                // Use HEAD for lightweight check or GET if HEAD is not supported by server
+                connection.setRequestMethod("HEAD");
+                connection.setConnectTimeout(6000);  // timeout in ms
+                connection.setReadTimeout(6000);
+
+                int responseCode = connection.getResponseCode();
+                if  (responseCode >= 200 && responseCode < 400) {
+                    String identifier = getCompactId(tokens);
+                    String prefix = StringUtils.substringBefore(identifier, ":").toLowerCase();
+
+                    if (result.get(prefix) == null) { //if the prefix is not in the compact ID, it must be in the previous token
+                        prefix = tokens[3].toLowerCase();
+                    }
+
+                    assertNotNull(result.get(prefix));
+                    Namespace dataType = (result.get(prefix) == null)
+                            ? result.get(StringUtils.substringAfter(prefix, "."))
+                            : result.get(prefix);
+                    assertNotNull(dataType);
+
+                }
+        }else {
+                assumeTrue(checkedNamespaces.contains("[" + checkedNamespace + "]"), "Namespace already checked, skipping");
+            }
+    }
+    public String getCompactId(String[] tokens){
+
+        String identifier;
+        if (tokens[tokens.length - 1].contains(":")){ //format : identifiers.org/[namespace prefix]:[accession]
+            identifier = tokens[tokens.length - 1];
+        } else if (tokens[tokens.length - 1].contains("[!\"#$%&'()*+,\\-./;<=>?@[\\\\\\]^_`{|}~]")){
```

Comment:
Code duplication! Why do I see the same code in two files! The getCompactId is duplicated and should just be imported. Probably same with other parts of this test code.
Remove code duplications.